pkg/piecestore/rpc/server/server.go

```diff
@@ -53,24 +56,83 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 	return server.Run(ctx)
 }
 
+//DirSize returns the total size of the files in that directory
+func DirSize(path string) (int64, error) {
+	var size int64
+	_, err := os.Stat(path)
+	if err != nil {
+		return 0, errors.New("path doesn't exists")
+	}
+	adjSize := func(_ string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			size += info.Size()
+		}
+		return err
+	}
+	err = filepath.Walk(path, adjSize)
+
+	return size, err
+}
+
 // Server -- GRPC server meta data used in route calls
 type Server struct {
-	DataDir string
-	DB      *psdb.DB
-	pkey    crypto.PrivateKey
+	DataDir        string
+	DB             *psdb.DB
+	pkey           crypto.PrivateKey
+	totalAllocated int64
 }
 
 // Initialize -- initializes a server struct
 func Initialize(ctx context.Context, config Config, pkey crypto.PrivateKey) (*Server, error) {
 	dbPath := filepath.Join(config.Path, "piecestore.db")
 	dataDir := filepath.Join(config.Path, "piece-store-data")
 
+	// read the allocated disk space from the config file
+	allocatedDiskSpace := config.AllocatedDiskSpace
+
+	// get the disk space details
+	diskSpace, err := disk.Usage("/")
+	freeDiskSpace := int64(diskSpace.Free)
+	if err != nil {
+		return nil, err
+	}
+
 	db, err := psdb.Open(ctx, dataDir, dbPath)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Server{DataDir: dataDir, DB: db, pkey: pkey}, nil
+	// get how much is currently used, if for the first time totalUsed = 0
+	totalUsed, err := DirSize(dataDir)
+	if err != nil {
+		//first time setup
+		totalUsed = 0
+	}
+
+	// check your hard drive is big enough
+	// first time setup as a piece node server
+	if (totalUsed == 0x00) && (freeDiskSpace < allocatedDiskSpace) {
+		return nil, errors.New("not enough space")
+	}
+
+	// on restarting the Piece node server, assuming already been working as a node
+	// used above the alloacated space, user changed the allocation space setting
+	// before restarting
+	if totalUsed >= allocatedDiskSpace {
+		log.Println("Warning!!! Used more space then allocated")
+		/** [TODO] any special handling needed here ... */
+		return &Server{DataDir: dataDir, DB: db, pkey: pkey, totalAllocated: allocatedDiskSpace}, nil
+	}
+
+	// the available diskspace is less than remaining allocated space,
+	// due to change of setting before restarting
+	if freeDiskSpace < (allocatedDiskSpace - totalUsed) {
+		log.Println("Warning!!! Disk space is less than remaining allocated space")
+		/** [TODO] any special handling needed here ... */
+		return &Server{DataDir: dataDir, DB: db, pkey: pkey, totalAllocated: allocatedDiskSpace}, nil
```

Comment:
let's set totalAllocated equal to freeDiskSpace

Reply:
done... also made the change to similar situation when the user decides to allocates less than what he already used then I set the allocated space to what is already used.